src/Dialect/ONNX/Rewrite.td

```diff
@@ -386,6 +386,13 @@ def FuseTransposePattern:  Pat<
   // Transpose with combined pattern.
   (ONNXTransposeOp $v, (CreateCombinedTransposedPattern $p1, $p2))>;
 
+def FuseTransposeAndLeakyReluPattern:  Pat<
+  // Transpose of LeakyRelu on a transpose.
+  (ONNXTransposeOp (ONNXLeakyReluOp (ONNXTransposeOp $v, $p1), $alpha), $p2),
+  // LeakyRelu on Transpose with combined pattern.
+  (ONNXTransposeOp (ONNXLeakyReluOp $v, $alpha, (returnType $v)),
+                   (CreateCombinedTransposedPattern $p1, $p2))>;
+
```

Comment:
You don't need to create `CombinedTransposedPattern` just swap transpose and leakyrelu, the two transposes will be combined automatically since we have a rule for that. For example,

`Transpose -> UnaryOp -> Transpose` becomes `UnaryOp -> Transpose -> Tranpose`.

Reply:
Thought so too... Initially though it did not hurt to do it here also.  One could imagine that if there are other rules with higher priority that would handle the transposes... it might be better indeed not to do it here.

Reply:
Yes, a global optimization is preferred to decide whether propagating transpose is beneficial or not.